internal/agent/device/applications/provider/app_handler.go

```diff
@@ -0,0 +1,110 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/flightctl/flightctl/internal/agent/client"
+	"github.com/flightctl/flightctl/internal/agent/device/applications/lifecycle"
+	"github.com/flightctl/flightctl/internal/agent/device/fileio"
+	"github.com/flightctl/flightctl/pkg/log"
+)
+
+type appTypeHandler interface {
+	// Dependencies returns the required applications required for the app to work
+	Dependencies() []string
```

Comment:
since dependencies are only used in verify does the interface need to expose this